src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
+            head = node;
+        } else {
+            Node<T> currentNode = tail;
+            currentNode.next = node;
+            node.prev = currentNode;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == STARTING_INDEX_OF_LIST) {
+            addFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            Node<T> node = new Node<>(value);
+            Node<T> currentNode = iterateToIndex(index);
+            node.next = currentNode;
+            node.prev = currentNode.prev;
+            currentNode.prev.next = node;
+            currentNode.prev = node;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = iterateToIndex(index);
+        if (currentNode == null) {
+            return null;
+        }
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = iterateToIndex(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == STARTING_INDEX_OF_LIST) {
+            Node<T> oldHead = head;
+            head = head.next;
+            size--;
+            return oldHead.value;
+        }
+        Node<T> currentNode = iterateToIndex(index);
+        unlink(currentNode);
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((currentNode.value == null && object == null)
+                    || (currentNode.value != null && currentNode.value.equals(object))) {
+                if (i == STARTING_INDEX_OF_LIST) {
+                    head = head.next;
+                    size--;
+                    return true;
+                }
+                unlink(currentNode);
+                return true;
+            } else {
+                currentNode = currentNode.next;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addFirst(T value) {
+        if (size == STARTING_INDEX_OF_LIST) {
+            add(value);
+        } else {
+            Node<T> node = new Node<>(value);
+            node.next = head;
+            head.prev = node;
+            head = node;
+            size++;
```

Comment:
you have alredy know that your node is fisrt. No needs to check it (you have a infinity loop here in some case)
```suggestion
            Node<T> node = new Node<>(value);
            node.next = head;
            head.prev = node;
            head = node;
            size++;
```

Reply:
wrote in Slack about this problem